docs/en/04_Changelogs/4.0.0.md

```diff
@@ -420,47 +185,13 @@ instead, or if used in an actual XML file use `.CDATA`.
 
 See the [Template casting](/developer_guides/templates/casting) section for specific details.
 
-### Automatically upgrading
-
-4.0 moves many classes to namespaces, and renames many methods.
-A tool is available to (semi-)automatically update your 3.x code to the new 4.0 names.
-
-See https://github.com/silverstripe/silverstripe-upgrader/ for the latest documentation on 
-how to upgrade your code. Typically the process will be:
-
-	
-	composer global require silverstripe/upgrader
-	cd ~/Project/Root
-	~/.composer/vendor/bin/upgrade-code upgrade ./mysite --write
-
-
-If you want to do a dry-run, omit the `--write` option to see a preview of a diff of
-all changed project files.
-
-This will resolve the majority of upgrading work, but for specific changes that will
-require manual intervention, please see the below upgrading notes.
-
-### Make sure templates are in correct locations
-
-Templates are now much more strict about their locations. You can no longer put a template in an arbitrary
-folder and have it be found. Case is now also checked on case-sensitive filesystems.
-
-Either include the folder in the template name (`renderWith('MyEmail.ss')` => `renderWith('emails/MyEmail.ss')`),
-move the template into the correct directory, or both.
-
-When using `<% include %>` template tag you should continue to leave out the `Includes` folder. 
-`<% include Sidebar %>` will match only match `Includes/Sidebar.ss`, not `Sidebar.ss`.
-Please refer to our [template syntax](/developer_guides/templates/syntax) for details.
-
-The `forms` template folder is no longer used to lookup templates for `FormField` instances.
-
-### Update code that uses SQLQuery
+#### Update code that uses SQLQuery
 
 Where your code once used SQLQuery you should now use SQLSelect in all cases, as this has been removed.
 
 See the [3.2.0](3.2.0) upgrading notes for details on how existing code should be upgraded.
 
-### New asset storage mechanism
+#### New asset storage mechanism
```

Comment:
Should mention that `file_get_contents()` and other direct filesystem operations should be avoided